pulsar-client/src/main/java/org/apache/pulsar/client/impl/PersistentAcknowledgmentsGroupingTracker.java

```diff
@@ -115,10 +115,11 @@ public PersistentAcknowledgmentsGroupingTracker(ConsumerImpl<?> consumer, Consum
      */
     @Override
     public boolean isDuplicate(@NonNull MessageId messageId) {
-        if (lastCumulativeAck.messageId == null) {
+        final MessageId messageIdOfLastAck = lastCumulativeAck.messageId;
```

Comment:
messageId is an reference Type. 

if you changed lastCumulativeAck.messageId to null, messageIdOfLastAck also will be null.

Reply:
Make sense. I'll fix it.

Reply:
agree

Reply:
I think the best way is to replace all the external access to `messageId` with a synchronized method of `LastCumulativeAck`, what do you think?

Reply:
maybe need a lock，when writing or reading the messageId